### PR TITLE
Add fast jet top tagger

### PR DESCRIPTION
**Pull request by lstroem**
BEGINRELEASENOTES
- Updated instructions on how to use FastJetUtil.h in stand-alone project.
- Moved getRecPars to FastJetUtil to enable broader use of conversion between PseudoJet objects and ReconstructedParticle (renamed convertFromPseudoJet).
- Added FastJet JHTopTagger implementation and made changes in other files to accommodate such tool implementations.

ENDRELEASENOTES

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,12 +16,17 @@ SET( ${PROJECT_NAME}_VERSION_PATCH 0 )
 
 ### DEPENDENCIES ############################################################
 
+# load fastjet util
```
Comment (andresailer) [inline CMakeLists.txt]: Please remove these commented lines

File: src/FastJetClustering.cc [hunk 2]
```diff
@@ -285,3 +285,22 @@ void FastJetClustering::end(){
     _rootfile->Close();
   }
 }
+
```
Comment (andresailer) [inline src/FastJetClustering.cc]: Please remove the commented lines

File: src/FastJetTopTagger.cpp [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+			     false);
+  
+  //Fastjet parameters
+  _fju->registerFastJetParameters( this );
+
+  //Top Tagger parameter
+  registerProcessorParameter("deltaP",
+			     "Subjets must carry at least this fraction of the original jet's p_t",
+			     _deltaP,
+			     std::string("0.05"));
```
Comment (andresailer) [inline src/FastJetTopTagger.cpp]: Why are you not defining `_deltaP`, `_deltaR`, `_cos_theta_W_max` as doubles?

**Review by andresailer — CHANGES_REQUESTED** [review]

File: include/FastJetTopTagger.h [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  std::string _R;
+  std::string _deltaP;
+  std::string _deltaR;
+  std::string _cos_theta_W_max;
+
+  // list of pseudo particles / jets
+  fastjet::JHTopTagger _jhtoptagger;
+
+ private:
+  FastJetTopTagger(const FastJetTopTagger& rhs);
```
Comment (andresailer) [inline include/FastJetTopTagger.h]:
```
FastJetTopTagger(const FastJetTopTagger& rhs) = delete;
FastJetTopTagger & operator = (const FastJetTopTagger&) = delete;
```
and remove the implementation of the copy constructor

File: include/FastJetTopTagger.h [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  int _statsFoundJets;
+  int _statsNrEvents;
+  int _statsNrSkippedEmptyEvents;
+  int _statsNrSkippedFixedNrJets;
+  int _statsNrSkippedMaxIterations;
+  bool _storeParticlesInJets;  
+
+  FastJetUtil* _fju;
+
+  std::string _R;
```
Comment (andresailer) [inline include/FastJetTopTagger.h:77]: Make them doubles

**Comment by lstroem** [comment]
The requested changes have all been implemented. Updated files pushed to my fork.

**Review by andresailer — APPROVED** [review]

**Pull request merged**